Library/Formula/node.rb

```diff
@@ -14,6 +14,7 @@ class Node < Formula
   end
 
   option "with-debug", "Build with debugger hooks"
+  option "with-icu4c", "Build with Intl (icu4c) support"
```

Comment:
It reflects this choice when configuring node:
https://github.com/nodejs/node#intl-ecma-402-support

I ported what we were doing for iojs here: https://github.com/aredridel/homebrew-iojs/blob/master/Formula/iojs.rb